import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/cstore/backup/BackupUploadService.java

```diff
@@ -0,0 +1,255 @@
+package com.google.cloud.healthcare.imaging.dicomadapter.cstore.backup;
+
+import com.google.cloud.healthcare.IDicomWebClient;
+import com.google.cloud.healthcare.IDicomWebClient.DicomWebException;
+import com.google.cloud.healthcare.imaging.dicomadapter.AetDictionary;
+import com.google.cloud.healthcare.imaging.dicomadapter.cstore.backup.BackupState;
+import com.google.cloud.healthcare.imaging.dicomadapter.cstore.backup.IBackupUploader;
+import com.google.cloud.healthcare.imaging.dicomadapter.cstore.multipledest.sender.CStoreSender;
+import com.google.cloud.healthcare.imaging.dicomadapter.monitoring.Event;
+import com.google.cloud.healthcare.imaging.dicomadapter.monitoring.MonitoringService;
+import com.google.cloud.healthcare.imaging.dicomadapter.cstore.backup.IBackupUploader.BackupException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BackupUploadService implements IBackupUploadService {
+
+  private final DelayCalculator delayCalculator;
+  private final IBackupUploader backupUploader;
+  private final ImmutableList<Integer> httpErrorCodesToRetry;
+  private final int attemptsAmount;
+
+  private Logger log = LoggerFactory.getLogger(this.getClass());
+
+  /**
+   * Create BackupUploadService instance.
+   * @param backupUploader DAO with simple write/read/remove operations.
+   * @param delayCalculator util class for reSend tasks schedule delay calculation.
+   */
+  public BackupUploadService(IBackupUploader backupUploader, Integer attemptsAmount, ImmutableList<Integer> httpErrorCodesToRetry,
+      DelayCalculator delayCalculator) {
+    this.backupUploader = backupUploader;
+    this.delayCalculator = delayCalculator;
+    this.httpErrorCodesToRetry = httpErrorCodesToRetry;
+    this.attemptsAmount = attemptsAmount;
+  }
+
+  @Override
+  public void createBackup(InputStream inputStream, String uniqueFileName) throws BackupException {
+    backupUploader.doWriteBackup(inputStream, uniqueFileName);
+    log.debug("sopInstanceUID={}, backup saved.", uniqueFileName);
+  }
+
+  @Override
+  public CompletableFuture startUploading(IDicomWebClient webClient, BackupState backupState) throws BackupException {
+    return scheduleUploadWithDelay(
+        backupState,
+        new HealthcareDestinationUploadAsyncJob(
+            webClient,
+            backupState),
+        0);
+  }
+
+  @Override
+  public CompletableFuture startUploading(CStoreSender cStoreSender, AetDictionary.Aet target, String sopInstanceUid, String sopClassUidBackupState,
```

Comment:
I suggest renaming sopClassUidBackupState to just sopClassUid to make it more clear what this is